src/Microsoft.OData.Edm/Validation/ExpressionTypeChecker.cs

```diff
@@ -486,6 +488,31 @@ private static bool TryCastBinaryConstantAsType(IEdmBinaryConstantExpression exp
             return true;
         }
 
+        private static bool TryCastEnumConstantAsType(IEdmEnumMemberExpression expression, IEdmTypeReference type, bool matchExactly, out IEnumerable<EdmError> discoveredErrors)
+        {
+            if (!type.IsEnum())
+            {
+                discoveredErrors = new EdmError[]
+                {
+                    new EdmError(expression.Location(),

```

Comment:
>Location [](start = 44, length = 8)

Is there any chance that expression may be null, leading to a null ref exception here (and below)? We cast to an IEdmEnumMemberExpression when calling this function; if the expression is not an IEdmEnumMemberExpression this will be null.  Is there any chance, when calling this function, that the expression will not be an IEdmEnumMemberExpression? Should we at least assert not-null, if not explicitly check for/handle the null case?